### PR TITLE
APIGOV-20991 - leave token locking and debug updates

### PR DESCRIPTION
**Pull request by sbolosan**
Based on iteration triage with Peter O.  deciding to leave token locking in

### CODE REVIEW

File: pkg/apic/auth/apicauth.go [hunk 1]
```diff
@@ -333,10 +336,9 @@ func (ptg *platformTokenGenerator) getPlatformTokens(requestToken string) (*axwa
 
 	defer closeHelper(resp.Body)
 	if resp.StatusCode != 200 {
-		body, err := ioutil.ReadAll(resp.Body)
-		if err != nil {
-			log.Debugf("bad response from AxwayID: %s: %s", resp.Status, body)
-		}
+		body, _ := ioutil.ReadAll(resp.Body)
+		log.Debugf("bad response from AxwayID: %s: %s, request token :%s, request time : %s", resp.Status, body, requestToken, startTime.String())
```
Comment (jcollins-axway) [inline pkg/apic/auth/apicauth.go:340]: Do we want to log the token here, is that not a security issue?

Comment (sbolosan) [reply pkg/apic/auth/apicauth.go:340]: oops... you're right.  this is left over from Peter's binary... thanks!

File: pkg/apic/auth/apicauth.go [hunk 2]
```diff
@@ -374,6 +376,7 @@ type platformTokenGetter struct {
 	*platformTokenGenerator
 	*keyReader
 	*tokenHolder
+	getTokenMutex *sync.Mutex
```
Comment (jcollins-axway) [inline pkg/apic/auth/apicauth.go:379]: My only issue with the mutex being here is if multiple processes create different token getters.  All we are locking is that the one instantiation is not requesting multiple at a time.  If the lock was a global in the package we can ensure that only one token getter can request a token at a time. Not sure if that matters or not, however.  @vivekschauhan thoughts?

Comment (vivekschauhan) [reply pkg/apic/auth/apicauth.go:379]: I think, locking on the instance should be good. Globally locking will lock each instance which might not be expected as instances can be configured with different configuration.

**Review by vivekschauhan — APPROVED** [review]

**Pull request merged**